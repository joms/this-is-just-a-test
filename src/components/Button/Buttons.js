import React from 'react';
import PropTypes from 'prop-types';

const Buttons = ({ children }) => <div className="buttons">{children}</div>;

Buttons.propTypes = {
    children: PropTypes.node.isRequired,
};

export default Buttons;
