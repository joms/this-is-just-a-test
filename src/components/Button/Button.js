import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import Buttons from './Buttons';

const Button = ({ children, invert, onClick, type, ...rest }) => {
    const classes = cx('button', { invert });

    return (
        <button className={classes} onClick={onClick} type={type} {...rest}>
            {children}
        </button>
    );
};

Button.defaultProps = {
    invert: false,
    onClick: () => null,
    rest: {},
    type: 'button',
};

Button.propTypes = {
    children: PropTypes.node.isRequired,
    invert: PropTypes.bool,
    onClick: PropTypes.func,
    rest: PropTypes.object,
    type: PropTypes.string,
};

Button.Buttons = Buttons;
export default Button;
