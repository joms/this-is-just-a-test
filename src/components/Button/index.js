import Button from './Button';
import Buttons from './Buttons';

export { Button, Buttons };

export default Button;
