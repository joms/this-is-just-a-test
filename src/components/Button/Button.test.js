import React from 'react';
import { shallow } from 'enzyme';
import Button from '.';

describe('<Button>', () => {
    const wrapper = shallow(<Button>Hello, world</Button>);
    it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders text', () => {
        expect(wrapper.text()).toEqual('Hello, world');
    });

    it('renders inverted', () => {
        const invertedWrapper = shallow(<Button invert>Hello, world</Button>);
        expect(invertedWrapper.hasClass('invert')).toBeTruthy();
    });

    it('clicks', () => {
        const onClick = jest.fn();
        const clickWrapper = shallow(<Button onClick={onClick}>Hello, world</Button>);
        clickWrapper.simulate('click');
        expect(onClick).toHaveBeenCalled();
    });
});
