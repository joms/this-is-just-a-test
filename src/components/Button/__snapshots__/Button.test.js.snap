// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> matches snapshot 1`] = `
<button
  className="button"
  onClick={[Function]}
  rest={Object {}}
  type="button"
>
  Hello, world
</button>
`;
